### ISSUE TITLE
gt_types after decomposition

### ISSUE THREAD

**Issue by muraveill** [arq5x/gemini#560]
Decomposing an heterozygous variant with two alts results in many "ref/." genotypes, which should probably be given a different gt_types than for "ref/alt" (1). 

Also it is not clear in the docs what these numbers mean ("A compressed binary vector of numeric genotype “types” (e.g., 0, 1, 2) - Inferred from the VCF GT genotype tag.", but they don't mean the same as in the VCF). That description could be improved. I read somewhere in the code that '2' actually means 'unknown/unknown', which is hard to guess.

**Comment by brentp** [comment]
the 0, 1, 2, 3 for HOM_REF, HET, UNKNOWN, HOM_ALT should not be visible to the user. The user should always use the HOM_REF, HET, etc constants. So you shouldn't have to guess anything.

Again, I agree that the decomposition does introduce some subtleties with "HEMIZYGOTES" but we think it's a good trade off.
The gts displayed when you request them should match those from the VCF. Are you seeing otherwise?

Not sure if this answers your concern, let me know if I can clarify more.

**Comment by delafont** [comment]
Yes, the gts do match the VCF. The problem arose when I queried the database directly with sql statements, filtering over the gt_types column, as described in [1]. The result was a rare variant with wrong HET genotypes in half of the samples (actually 'hemizygotes'). I am not sure what this column is used for, but as it is, 
1. it is wrong for a user to use it directly, 
2. if it is used internally for API queries, it may cause the same kind of errors, so maybe worth signalling. I guess if you ask for HET through API you also get "hemizygotes" in the result, as it will search for '1' in that column.
Anyway the description [1] for that field could be improved.

[1] http://gemini.readthedocs.org/en/latest/content/database_schema.html

**Comment by muraveill** [comment]
Actually it is a dup of https://github.com/arq5x/gemini/issues/510

**Comment by ddkinnamon** [comment]
I have a large enough data set that I have to deal with lots of hemizygous decomposed variants, and it has been a real headache. It looks like I could reliably distinguish hemizygotes from heterozygotes when gt_types==HET by using gt_alt_depths (0 is hemizygote, >0 is a true heterozygote). Unfortunately, it's unclear to me how I could do this within a GEMINI query, especially when I am using --family-wise to try to get variants that segregate among affected family members. Any suggestions?

**Comment by brentp** [comment]
Agreed, this is becoming more important as people get larger sample sizes. This is a weakness of how we do things now. I think it's possible using alt_depths, as you suggest, but the queries would get quite complex. we are thinking about how to address this but don't have a good solution as of yet. The thoughts are in #510;I think we will need a different internal representation.
